packages/query/src/stacks/stacks-client.ts

```diff
@@ -0,0 +1,305 @@
+      return resp.data;
+    },
+  };
+}
+
+function HiroTokensApi(basePath: string) {
+  const rateLimiter = getHiroApiRateLimiter(basePath);
+  return {
+    async getFtMetadata(address: string) {
+      const resp = await rateLimiter.add(
+        () => axios.get<FtMetadataResponse>(`${basePath}/metadata/v1/ft/${address}`),
+        { throwOnTimeout: true }
+      );
+      return resp.data;
+    },
+    async getNftMetadata(address: string, tokenId: number) {
+      const resp = await rateLimiter.add(
+        () => axios.get<NftMetadataResponse>(`${basePath}/metadata/v1/nft/${address}/${tokenId}`),
+        { throwOnTimeout: true }
+      );
+      return resp.data;
+    },
+  };
+}
+
+export interface CallReadOnlyFunctionArgs {
+  contractAddress: string;
+  contractName: string;
+  functionName: string;
+  readOnlyFunctionArgs: ReadOnlyFunctionArgs;
+  tip?: string;
+  signal?: AbortSignal;
+}
+
+function HiroSmartContractsApi(basePath: string) {
+  const rateLimiter = getHiroApiRateLimiter(basePath);
+  return {
+    async callReadOnlyFunction({
+      contractAddress,
+      contractName,
+      functionName,
+      readOnlyFunctionArgs,
+      tip,
+      signal,
+    }: CallReadOnlyFunctionArgs) {
+      const resp = await rateLimiter.add(
+        () =>
+          axios.post<ReadOnlyFunctionSuccessResponse>(
+            `${basePath}/v2/contracts/call-read/${contractAddress}/${contractName}/${functionName}?tip=${tip}`,
+            {
+              ...readOnlyFunctionArgs,
+              signal,
+            }
+          ),
+        { signal, throwOnTimeout: true }
+      );
+      return resp.data;
+    },
+    async getContractInterface(
+      contractAddress: string,
+      contractName: string,
+      signal?: AbortSignal
+    ) {
+      const resp = await rateLimiter.add(
+        () =>
+          axios.get<ContractInterfaceResponseWithFunctions>(
+            `${basePath}/v2/contracts/interface/${contractAddress}/${contractName}`,
+            {
+              signal,
+            }
+          ),
+        { throwOnTimeout: true }
+      );
+      return resp.data;
+    },
+  };
+}
+
+function Stx20Api() {
+  const url = STX20_API_BASE_URL_MAINNET;
+
+  return {
+    async getStx20Balances(address: string) {
+      const resp = await axios.get<Stx20BalanceResponse>(`${url}/balance/${address}`);
+      return resp.data.balances;
+    },
+  };
+}
+
+export interface StacksClient {
+  hiroAccountsApi: ReturnType<typeof HiroAccountsApi>;
+  hiroFeesApi: ReturnType<typeof HiroFeesApi>;
+  hiroInfoApi: ReturnType<typeof HiroInfoApi>;
+  hiroNamesApi: ReturnType<typeof HiroNamesApi>;
+  hiroNetworkApi: ReturnType<typeof HiroNetworkApi>;
+  hiroNonFungibleTokensApi: ReturnType<typeof HiroNonFungibleTokensApi>;
+  hiroSmartContractsApi: ReturnType<typeof HiroSmartContractsApi>;
+  hiroTokensApi: ReturnType<typeof HiroTokensApi>;
+  hiroTransactionsApi: ReturnType<typeof HiroTransactionsApi>;
+  stx20Api: ReturnType<typeof Stx20Api>;
+}
```

Comment:
I feel like the grouping of these comes from the old way of doing it with the generated client. Why not just have `client.getBnsNames` vs having to access via a nested object? DX is better to have a long list because I can search/browse all available methods.